torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -3119,9 +3121,16 @@ at::Tensor XLANativeFunctions::upsample_bilinear2d(
   std::vector<int64_t> scaled_output_size =
       torch::lazy::ToVector<int64_t>(output_size);
   if ((scales_h && *scales_h != 1.0) || (scales_w && *scales_w != 1.0)) {
-    scaled_output_size = GetOutputSizeWithScale(input_dims, scales_h, scales_w,
-                                                scaled_output_size);
-    if (!output_size.empty()) {
+    scaled_output_size = GetOutputSizeWithScale(
+        input_dims, scales_h, scales_w, scaled_output_size,
+        /*round_with_scale_factor=*/false);
+  }
+  if (!output_size.empty()) {
+    if (!(scaled_output_size.at(0) == output_size.at(0) &&
+          scaled_output_size.at(1) == output_size.at(1))) {
+      scaled_output_size = GetOutputSizeWithScale(
+          input_dims, scales_h, scales_w, scaled_output_size,
+          /*round_with_scale_factor=*/true);
```

Comment:
Seems like that `}` on the line 3127 changes a bit the logic when the check `scaled_output_size` vs `output_size` happens.

Reply:
Thanks for the catch, I've moved this code into the `if` statement to keep the existing behavior. Let me know if that makes more sense.